common/service.go

```diff
@@ -19,6 +19,8 @@ type Service interface {
 	IsRunning() bool
 
 	String() string
+
+	SetLogger(log.Logger)
```

Comment:
they should not. but in single cases where we log something in a constructor, there is no other way